blocks/bricks/parallel.py

```diff
@@ -261,3 +263,31 @@ def apply_inputs(self):
     @apply.property('outputs')
     def apply_outputs(self):
         return self.target_names
+
+
+class Merge(Parallel):
+    """Merges several variables by applying a transformation and summing.
+
+    input_names : list of str
+        The input names.
+    input_dims : dict
+        The dictionary of input dimensions, keys are input names, values
+        are dimensions.
+    output_dim : int
+        The output dimension of the merged variables.
+    prototype : :class:`~blocks.bricks.Feedforward`
+        A transformation prototype. A copy will be created for every
+        input.  If ``None``, a linear transformation without bias is used.
+    child_prefix : str, optional
+        A prefix for children names. By default "transform" is used.
+
```

Comment:
When you not specify attributes, that is not explain how to configure your class after the construction, you effectively make it non-lazy only.

I guess we should add a property `output_dim`, and document this property and the `input_dims` dictionary as attributes.